### PR TITLE
Dtsstci-398

### PR DESCRIPTION
**Pull request by Jefex7**
### Change description ###

Audit Unit Tests in Case API - SystemUpdate Module Service package

### JIRA link (if applicable) ###

[https://tools.hmcts.net/jira/browse/DTSSTCI-398](url)

**Before merging a pull request make sure that:**

- [x] tests have been updated / new tests has been added (if needed)
- [x] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: cftlib_log_file_IS_UNDEFINED [hunk 1]
```diff
@@ -0,0 +1 @@
+2024-01-23T01:36:15.903 DEBUG [main] o.springframework.boot.devtools.restart.Restarter Creating new Restarter for thread Thread[main,5,main]
```
Comment (gmmagruder) [inline cftlib_log_file_IS_UNDEFINED:1]: Why is this log file here?

Comment (Jefex7) [reply cftlib_log_file_IS_UNDEFINED]: removed my bad

File: src/main/java/uk/gov/hmcts/sptribs/systemupdate/service/CcdSearchService.java [hunk 2]
```diff
@@ -37,8 +37,6 @@ public class CcdSearchService {
     @Autowired
     private CoreCaseDataApi coreCaseDataApi;
 
-
-
     public List<CaseDetails> searchForAllCasesWithQuery(
         final BoolQueryBuilder query, User user, String serviceAuth, final State... states) {
```
Comment (rajaram-nagarajan) [inline src/main/java/uk/gov/hmcts/sptribs/systemupdate/service/CcdSearchService.java:42]: Method searchForCasesWithQuery can be converted to private and moved to last in the file.

File: src/main/java/uk/gov/hmcts/sptribs/systemupdate/service/CcdSearchService.java [hunk 3]
```diff
@@ -37,8 +37,6 @@ public class CcdSearchService {
...
```
Comment (rajaram-nagarajan) [inline src/main/java/uk/gov/hmcts/sptribs/systemupdate/service/CcdSearchService.java:41]: There is a dummy/test case reference 1695290772211061L added in the java code under the method searchForCasesWithVersionLessThan. This has to be removed and the corresponding test needs to be adjusted as required.

File: src/main/java/uk/gov/hmcts/sptribs/systemupdate/service/CcdSearchService.java [hunk 4]
```diff
@@ -37,8 +37,6 @@ public class CcdSearchService {
     @Autowired
     private CoreCaseDataApi coreCaseDataApi;
 
-
-
     public List<CaseDetails> searchForAllCasesWithQuery(
```
Comment (rajaram-nagarajan) [inline src/main/java/uk/gov/hmcts/sptribs/systemupdate/service/CcdSearchService.java:40]: The parameter latestVersion wasn't used inside the method searchForCasesWithVersionLessThan. Can this be removed?

File: src/main/java/uk/gov/hmcts/sptribs/systemupdate/service/CcdSearchService.java [hunk 5]
```diff
@@ -37,8 +37,6 @@ public class CcdSearchService {
     @Autowired
     private CoreCaseDataApi coreCaseDataApi;
 
-
-
```
Comment (rajaram-nagarajan) [inline src/main/java/uk/gov/hmcts/sptribs/systemupdate/service/CcdSearchService.java:41]: Line no: 96 - toString method can be removed as it is redundant.

**Review by rajaram-nagarajan — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/sptribs/systemupdate/service/CcdSearchService.java (hunk 4 above)
Comment (Jefex7) [reply src/main/java/uk/gov/hmcts/sptribs/systemupdate/service/CcdSearchService.java:40]: latestVersion seems to be used in BaseMigration & BaseMigrationTest so I changed the SearchSourcebuilder query to use latestVersion Can you check please if that's correct or not.

Comment (rajaram-nagarajan) [reply src/main/java/uk/gov/hmcts/sptribs/systemupdate/service/CcdSearchService.java:40]: dataVersion field has been retired. And don't know how else we can check the latest version. So, let's leave this like this. This method may need to be removed in future.

**Review by rajaram-nagarajan — APPROVED** [review]
LGTM

**Review by tomxelliott — APPROVED** [review]

**Pull request merged**